error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1904,4 +1905,33 @@ Duration after(StepVerifier.LastStep step, Duration duration) {
       return step.verifyTimeout(duration);
     }
   }
+
+  /** Prefer {@link Mono#fromFuture(Supplier)} over {@link Mono#fromFuture(CompletableFuture)}. */
+  static final class MonoFromFuture<T> {
+    @BeforeTemplate
+    Mono<T> before(CompletableFuture<T> future) {
+      return Mono.fromFuture(future);
+    }
+
+    @AfterTemplate
+    Mono<T> after(CompletableFuture<T> future) {
+      return Mono.fromFuture(() -> future);
+    }
+  }
+
+  /**
+   * Prefer {@link Mono#fromFuture(Supplier, boolean)} over {@link
+   * Mono#fromFuture(CompletableFuture, boolean)}.
+   */
+  static final class MonoFromFutureSuppressCancel<T> {
```

Comment:
@Stephan202 according to our undocumented naming scheme, we would now call this `MonoFromFutureBoolean` right?

Reply:
Strictly speaking even `MonoFromFutureSupplierBoolean` :smile: